table/meta_blocks.cc

```diff
@@ -172,6 +173,8 @@ Status ReadProperties(const Slice &handle_value, RandomAccessFile *file,
       {TablePropertiesNames::kFixedKeyLen,
        &new_table_properties->fixed_key_len}, };
 
+  new_table_properties->compression_type = found_compression_type;
```

Comment:
some blocks in the table might be compressed, while others might not be (because the compression ratio for those blocks is not worth the trouble). it might be better to add an if around this:

    if (found_compression_type != kNoCompression) {
            new_table_properties->compression_type = found_compression_type; 
   }